### PR TITLE
BXMSPROD-496 - Eliminate branching mapping file

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.jboss.org/browse/BXMSPROD-496

### CODE REVIEW

**Review by mbiarnes — COMMENTED** [review]
LGTM - albeit I am not able to see how the PRs will work after this. 
And I don't know what will happen the next time we will branch to a new branch and we don't use anymore https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/master/script/branch-mapping.yaml

File: src/main/java/org/kie/jenkinsci/plugins/kieprbuildshelper/RepositoryLists.java [hunk 1]
```diff
@@ -70,7 +64,9 @@ private static URL createUrlForRepositoryList(GitHubRepository repo, GitBranch b
      * TODO: this is an ugly hack. The dependency between repositories (or directly modules) should to be checked automatically for every build
      */
     public static List<Tuple<GitHubRepository, GitBranch>> filterOutUnnecessaryRepos(List<Tuple<GitHubRepository, GitBranch>> repos, GitHubRepository baseRepo) {
-        // nothing depends on stuff from -tools repo
+        // Errai is not upstream repository anymore
+    	repos.removeIf(repo -> repo._1().equals(new GitHubRepository("errai", "errai")));
```
Comment (almope) [inline src/main/java/org/kie/jenkinsci/plugins/kieprbuildshelper/RepositoryLists.java]: Is this needed? I think after removing the `branch-mapping.yaml` file the only source for repos is the droolsjbpm-build-`bootstrap/scripts/repository-list.txt` file where `errai` is not listed

**Review by almope — APPROVED** [review]
Apart from my previous comment about `errai` filtering the change looks excellent. @mareknovotny thanks for picking this one!

File: src/main/java/org/kie/jenkinsci/plugins/kieprbuildshelper/RepositoryLists.java (hunk 1 above)
Comment (mareknovotny) [reply src/main/java/org/kie/jenkinsci/plugins/kieprbuildshelper/RepositoryLists.java]: no, that was some tryout ;)

**Comment by mareknovotny** [comment]
the main work will come with testing it on running instance of Jenkins, so this part was easy, the harder will come ;)

**Pull request merged**